pkg/mssqlcontainer/ingest/ingest.go

```diff
@@ -0,0 +1,135 @@
+package ingest
+
+import (
+	"fmt"
+	"github.com/microsoft/go-sqlcmd/internal/container"
+	"github.com/microsoft/go-sqlcmd/internal/uri"
+	"github.com/microsoft/go-sqlcmd/pkg/mssqlcontainer/ingest/extract"
+	"github.com/microsoft/go-sqlcmd/pkg/mssqlcontainer/ingest/location"
+	"github.com/microsoft/go-sqlcmd/pkg/mssqlcontainer/ingest/mechanism"
+	"path/filepath"
+	"strings"
+)
+
+type ingest struct {
+	uri         uri.Uri
+	location    location.Location
+	controller  *container.Controller
+	mechanism   mechanism.Mechanism
+	options     mechanism.BringOnlineOptions
+	extractor   extract.Extractor
+	containerId string
+	query       func(text string)
+}
+
+func (i *ingest) IsExtractionNeeded() bool {
+	i.extractor = extract.NewExtractor(i.uri.FileExtension(), i.controller)
+	if i.extractor == nil {
+		return false
+	} else {
+		return true
+	}
```

Comment:
I propose  we pick one of three styles to avoid superfluous "else" blocks

1:
```go
return i.extractor != nil
```

2.
```go
func (i *ingest) IsExtractionNeeded() needed bool {
i.extractor = extract.NewExtractor(i.uri.FileExtension(), i.controller)
needed = iExtractor != nil
}
```

3. 
```go
func (i *ingest) IsExtractionNeeded() bool {
 i.extractor = extract.NewExtractor(i.uri.FileExtension(), i.controller)
 if i.extractor == nil {
  return false
 }
  return true
}
```

My vote is 1 .